gitlab/cli.py

```diff
@@ -126,7 +132,86 @@ def _get_base_parser(add_help=True):
         ),
         required=False,
     )
-
+    # TODO: deduplicate everything by providing some data struct to loop over
+    parser.add_argument(
+        "--url",
+        help=("GitLab server URL [env var: GITLAB_URL]"),
+        required=False,
+        default=os.getenv("GITLAB_URL"),
+    )
+    parser.add_argument(
```

Comment:
On many systems, command-line arguments passed to programs can be seen by all other users of the system. I'm not convinced it's a good approach to facilitate the passing of secret tokens using CLI arguments like this

Reply:
This was mainly added for completeness and to be consistent with many other CLI tools that do offer it, including gitlab's own tooling https://gitlab.com/gitlab-org/release-cli/-/blob/master/docs/index.md#using-this-tool-with-a-private-token). Seems like even HashiCorp's vault allows login via CLI args.

But I agree otherwise and we already suggest using credential helpers for tokens in the docs. Maybe a compromise would be for `gitlab` to issue a warning if it finds that credentials were passed via CLI args + a mention in the docs advising against it.